### DISCUSSION TITLE
Bug: SearchFilter renders search field with invalid value

### DISCUSSION THREAD

**Discussion by drewg3r** [encode/django-rest-framework#9018] [Potential Issue] [answered]
## Bug Description

When using the `SearchFilter` with multiple words in the search query, the rendered search field only displays the first word instead of the full search query.

## Steps to Reproduce

1. Set up a `ListAPIView` with `filter_backends = [filters.SearchFilter]` ([example from documentation](https://www.django-rest-framework.org/api-guide/filtering/#searchfilter)).
2. Access the Browsable API and navigate to the ListAPIView.
3. Click on the "Filters" button.
4. Fill in the search fields with any two words separated by a space.
5. Click the "Search" button.
6. After the page refreshes, click the "Filters" button again.
7. Observe that the search field now only displays the first word of the search query, while the search is actually performed over the full text of the query.

## Expected Behavior

The search field should display the full search query entered by the user.

## Actual Behavior

The search field only displays the first word of the search query.

## Probable cause of the bug

The issue may lie in the filters.py file, specifically [line 143](https://github.com/encode/django-rest-framework/blob/8dd4250d0234ddf6c6a19a806639678ef7786468/rest_framework/filters.py#L143).

**Comment by auvipy** [comment]
can you come with a fix which contain unit tests?

**Reply by drewg3r** [reply, marked as answer]
Yes, I have created a PR #9023
